src/citrine/waiting/waiting.py

```diff
@@ -0,0 +1,77 @@
+from time import time, sleep
+from typing import Callable
+from pprint import pprint
+from citrine._rest.collection import Collection
+from citrine.resources.workflow_executions import (
+    WorkflowExecution,
+    WorkflowExecutionStatus,
+)
+
+
+def _wait_until(condition: Callable[[], bool], timeout: float = 900.0, interval: float = 3.0):
+    """Poll until the provided condition is truthy or the timeout is reached."""
+    start = time()
+    while not condition() and time() - start < timeout:
+        sleep(interval)
+    assert condition(), "Timeout reached, but condition is still False."
```

Comment:
I don't think we want to be throwing an assertion error here.  I think a `RuntimeError` would be more appropriate.  It would be particularly good to define our own error subclass (could be just at the top of this file) and throw that so that callers can catch that error specifically.  Here is [an example](https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/builders/descriptors.py#L15) of that pattern.